test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/BiPredicateTest.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.function
+
+import java.util.function.BiPredicate
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class BiPredicateTest {
+  import BiPredicateTest._
+
+  private val largerThan10LessThan30 = makeBiPredicate[Int, Int](_ > 10, _ < 30)
+  private val even = makeBiPredicate[Int, Int](_ % 2 == 0, _ % 2 == 0)
+
+  private val throwingPredicate =
+    makeBiPredicate[Any, Any](x => throw new ThrowingPredicateException(x), _ => true)
+
+  private val dontCallPredicate =
+    makeBiPredicate[Any, Any](x => throw new AssertionError(s"dontCallPredicate.test($x)"), _ => true)
+
+  @Test def and(): Unit = {
+    // Truth table
+    val evenAndInRange = largerThan10LessThan30.and(even)
+
+    // In Both Range and Even
+    assertTrue(evenAndInRange.test(22, 22))
+    assertTrue(evenAndInRange.test(40, 22))
+
+    // In Both Range and Not Even
+    assertFalse(evenAndInRange.test(41, 22)) // Left Odd
+    assertFalse(evenAndInRange.test(42, 21)) // Right Odd
+    assertFalse(evenAndInRange.test(41, 21)) // Both Odd
+
+    // In One Range and Even
+    assertFalse(evenAndInRange.test(6, 6))   // Left Range false
+    assertFalse(evenAndInRange.test(42, 42)) // Right Range false
+    assertFalse(evenAndInRange.test(6, 42))  // Both Range false
```

Comment:
I don't think it makes sense to test all conditions on the bi-predicate itself: There is no code to be tested, it merely tests the implementation of `mkBiPredicate`.

Reply:
I was up in the air about this myself.  I erred on the side to try and be more exhaustive than not, even if redundant. But in this case the redundancy may just add more confusion to the test/maintenance/etc.  I'll do another pass through the tests and see if I can simplify.